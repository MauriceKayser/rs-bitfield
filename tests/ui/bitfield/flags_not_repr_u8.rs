//! Assertion failure: _FLAGS_IN_FIELD_0_MUST_BE_REPR_U8

extern crate alloc;

#[bitfield::bitfield(16)]
struct BitField(Flags);

#[derive(Copy, Clone, Debug)]
#[repr(u16)] // Must be `u8` instead of `u16`.
enum Flags {
    F0,
    F1,
    F3 = 3
}

impl Flags {
    #[allow(unused)]
    #[inline(always)]
    const fn iter() -> &'static [Self] {
        &[Self::F0, Self::F1, Self::F3]
    }

    #[allow(unused)]
    #[inline(always)]
    const fn max() -> Self {
        let mut i = 0;
        let mut max = Flags::F0;

        while i < Self::iter().len() {
            let current = Self::iter()[i];
            if current as u8 > max as u8 {
                max = current;
            }

            i += 1;
        }

        max
    }
}

fn main() {}