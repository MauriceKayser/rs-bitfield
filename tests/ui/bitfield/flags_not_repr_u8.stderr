error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/flags_not_repr_u8.rs:6:17
  |
6 | struct BitField(Flags);
  |                 ^^^^^ the evaluated program panicked at 'Flags must be #[repr(u8)]', $DIR/tests/ui/bitfield/flags_not_repr_u8.rs:6:17
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
